sympy/functions/elementary/integers.py

```diff
@@ -131,14 +131,23 @@ def _eval_number(cls, arg):
         if arg.is_NumberSymbol:
             return arg.approximation_interval(Integer)[0]
 
+    def _eval_as_leading_term(self, x, cdir=0):
+        arg = self.args[0]
+        arg0 = arg.subs(x, 0)
+        if arg0.is_integer:
+            return arg0
+        elif arg0.is_finite:
+            return self.func(arg0)
+        return arg
```

Comment:
We should take the leading term of `arg`. For example, the leading term of `floor(1/sin(x))` should be `1/x`